### ISSUE TITLE
[Bug]: Logging in via Google as banned user results in a crash

### ISSUE THREAD

**Issue by brezinajn** [supabase-community/supabase-kt#432]
### General Info

- [X] I checked for similar bug report
- [X] I am using the latest version
- [X] I checked the [troubleshooting](https://github.com/supabase-community/supabase-kt/wiki/Troubleshooting) page for similar problems
- [X] I enabled [logging](https://github.com/supabase-community/supabase-kt/wiki/Logging) and checked the logs

### Version(s)

2.0.4

### Target(s)

Android

### What happened? (include your code)

Logging in as a banned user via Google provider, results in the API call response with 500. That's for sure a Supabase issue. But there should probably be a mechanism to handle such exceptions and prevent them bubbling up and crashing the app.

### Steps To Reproduce (optional)

Ban a user via auth.adminApi
Log in as the banned user

### Relevant log output (optional)

```shell
Fatal Exception: io.github.jan.supabase.exceptions.BadRequestRestException: server_error (Internal Server Error)
URL: https://redacted.supabase.co/auth/v1/token?grant_type=id_token
Headers: redacted
Http Method: POST
       at io.github.jan.supabase.gotrue.AuthImpl.parseErrorResponse(AuthImpl.kt:489)
       at io.github.jan.supabase.gotrue.AuthenticatedSupabaseApiKt$authenticatedSupabaseApi$3.invoke(AuthenticatedSupabaseApi.kt:58)
       at io.github.jan.supabase.gotrue.AuthenticatedSupabaseApiKt$authenticatedSupabaseApi$3.invoke(AuthenticatedSupabaseApi.kt:58)
       at io.github.jan.supabase.network.SupabaseApi.rawRequest$suspendImpl(SupabaseApi.kt:25)
       at io.github.jan.supabase.network.SupabaseApi$rawRequest$1.invokeSuspend(:15)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
       at androidx.compose.ui.platform.AndroidUiDispatcher.performTrampolineDispatch(AndroidUiDispatcher.android.kt:81)
       at androidx.compose.ui.platform.AndroidUiDispatcher.access$performTrampolineDispatch(AndroidUiDispatcher.android.kt:41)
       at androidx.compose.ui.platform.AndroidUiDispatcher$dispatchCallback$1.run(AndroidUiDispatcher.android.kt:57)
       at android.os.Handler.handleCallback(Handler.java:958)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:205)
       at android.os.Looper.loop(Looper.java:294)
       at android.app.ActivityThread.main(ActivityThread.java:8248)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```

**Comment by jan-tennert** [comment]
@brezinajn So how do you log in? Via Compose Auth?

**Comment by brezinajn** [comment]
Yes via Compose Auth.

```Kotlin
// client creation
install(Auth)
install(ComposeAuth) {
    googleNativeLogin(serverClientId = serverClientId)
}
...
// use site
composeAuth.rememberSignInWithGoogle()
```

**Comment by jan-tennert** [comment]
> Yes via Compose Auth.
> 
> ```kotlin
> // client creation
> install(Auth)
> install(ComposeAuth) {
>     googleNativeLogin(serverClientId = serverClientId)
> }
> ...
> // use site
> composeAuth.rememberSignInWithGoogle()
> ```

Okay, does this crash happen on Android 14? (I'm asking because it seems like there is a try catch missing for the Android 14 implementation)

**Comment by brezinajn** [comment]
Yes

**Comment by jan-tennert** [comment]
Alright, a fix will be included in the upcoming 2.1.0 beta.
